demo/estimate_smpl.py

```diff
@@ -0,0 +1,498 @@
+
+        for i, img_i in enumerate(sorted(os.listdir(frames_folder))):
+            for person_i in track_ids_lists[i]:
+                body_pose_.append(smpl_poses[i][person_i][1:])
+                global_orient_.append(smpl_poses[i][person_i][:1])
+                smpl_betas_.append(smpl_betas[i][person_i])
+                verts_.append(verts[i][person_i])
+                pred_cams_.append(pred_cams[i][person_i])
+                bboxes_xyxy_.append(bboxes_xyxy[i][person_i])
+                image_path_.append(os.path.join('images', img_i))
+                person_id_.append(person_i)
+                frame_id_.append(frame_id_list[i])
+
+        smpl = {}
+        smpl['body_pose'] = np.array(body_pose_).reshape((-1, 23, 3))
+        smpl['global_orient'] = np.array(global_orient_).reshape((-1, 3))
+        smpl['betas'] = np.array(smpl_betas_).reshape((-1, 10))
+        human_data['smpl'] = smpl
+        human_data['verts'] = verts_
+        human_data['pred_cams'] = pred_cams_
+        human_data['bboxes_xyxy'] = bboxes_xyxy_
+        human_data['image_path'] = image_path_
+        human_data['person_id'] = person_id_
+        human_data['frame_id'] = frame_id_
+        human_data.dump(osp.join(args.output, 'inference_result.npz'))
+
+    # To compress vertices array
+    compressed_verts = np.zeros([frame_num, max_instance, 6890, 3])
+    compressed_cams = np.zeros([frame_num, max_instance, 3])
+    compressed_bboxs = np.zeros([frame_num, max_instance, 5])
+
+    for i, track_ids_list in enumerate(track_ids_lists):
+        instance_num = len(track_ids_list)
+        compressed_verts[i, :instance_num] = verts[i, track_ids_list]
+        compressed_cams[i, :instance_num] = pred_cams[i, track_ids_list]
+        compressed_bboxs[i, :instance_num] = bboxes_xyxy[i, track_ids_list]
+
+    assert len(frame_id_list) > 0
+
+    if args.show_path is not None:
+        if args.output is not None:
+            frames_folder = os.path.join(args.output, 'images')
+        else:
+            frames_folder = osp.join(Path(args.show_path).parent, 'images')
+            os.makedirs(frames_folder, exist_ok=True)
+            array_to_images(
+                np.array(frames_iter)[frame_id_list],
+                output_folder=frames_folder)
+        body_model_config = dict(model_path=args.body_model_dir, type='smpl')
+        visualize_smpl_hmr(
+            verts=compressed_verts,
+            cam_transl=compressed_cams,
+            bbox=compressed_bboxs,
+            output_path=args.show_path,
+            render_choice=args.render_choice,
+            resolution=frames_iter[0].shape[:2],
+            origin_frames=frames_folder,
+            body_model_config=body_model_config,
+            overwrite=True,
+            palette=args.palette,
+            read_frames_batch=True)
+
+
+def main(args):
+
+    # prepare input
+    frames_iter = prepare_frames(args.input_path)
```

Comment:
[mmcv.VideoReader](https://github.com/open-mmlab/mmcv/blob/222f38075b8dfd5f11b56b7918f67fd983de7e28/mmcv/video/io.py#L67) is a wrapper around cv2.VideoCapture. In demo_utils.py, we should be able to support more than '.mp4'.

Reply:
done